src/actions/PetitionsActions.js

```diff
@@ -9,24 +9,35 @@ import {
   UPDATE_CURRENT_CITY
 } from './actionTypes';
 
-export function fetchPetitions ({ location, params }) {
-  // Get query from react-router locatiin
+export function fetchPetitions ({ location, params = {} }) {
+  // Get query from react-router location
   const { query } = location;
 
-  // Construct our query params, based on
-  // route params or query string params
+  // Construct our query params based on route params or query string params
   const queryParams = {
-    page: parseInt(params && params.page || query.page || 1),
-    city: params && params.city || query.city || '',
-    cityName: params && params.cityName || query.cityName || '',
-    limit: parseInt(query.limit || 12)
+    page: parseInt(params.page || query.page || 1),
+    city: params.city || query.city || '',
+    cityName: params.cityName || query.cityName || '',
+    state: params.state || query.state || '',
+    limit: parseInt(query.limit || 12),
+    sort: query.sort || ''
   };
 
-  // Take any query string values and encode them,
-  // picking the relavent props for filering
+  // To avoid repetition between URL params and query params, we don’t save
+  // params taken from URL.
+  let encodedParams = [];
+
+  for (const key in queryParams) {
+    if (query[key]) {
+      encodedParams.push(key);
+    }
+  }
```

Comment:
why can't we just use `pick` below here?
we are in control of what we want to use as real params vs query params? e.g. `['limit', 'sort']` are the query params?